libsolidity/codegen/YulUtilFunctions.cpp

```diff
@@ -517,6 +517,108 @@ string YulUtilFunctions::arrayLengthFunction(ArrayType const& _type)
 	});
 }
 
+std::string YulUtilFunctions::resizeDynamicArrayFunction(ArrayType const& _type)
+{
+	solAssert(_type.location() == DataLocation::Storage, "");
+	solAssert(_type.isDynamicallySized(), "");
+	solUnimplementedAssert(!_type.isByteArray(), "Byte Arrays not yet implemented!");
+
+	if (!_type.isByteArray() && _type.baseType()->storageBytes() < 32)
+		solAssert(_type.baseType()->isValueType(), "Invalid storage size for non-value type.");
+
+	string functionName = "resize_array_" + _type.identifier();
+	return m_functionCollector->createFunction(functionName, [&]() {
+		return Whiskers(R"(
+			function <functionName>(array, newLen) {
+				let oldLen := <fetchLength>(array)
+
```

Comment:
I think this should have a length restriction. Doesn't the old code have one?

Reply:
I don't think the old code had one: https://github.com/ethereum/solidity/blob/develop/libsolidity/codegen/ArrayUtils.cpp#L646

What length do we want to restrict it to?

Reply:
The old code did not, but all the abi encoding code does and there is also a restriction on statically-sized array. Maybe we should use the same restriction.

On the other hand, we want to disable explicitly setting the length anyway.

Reply:
Isn't it way more work and less efficient to set an array length using push/pop loops?

Reply:
Do you have examples? The idea behind disallowing it is that people do not realize the extra work behind shortening an array.

Reply:
hmm no examples, just my intuition.. I am still fairly unaware of typical solidity use cases